### PR TITLE
Mat 3911 button for test execution

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-3911](https://jira.cms.gov/browse/MAT-3911)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [x] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Comment by sb-cecilialiu** [comment]
NOTE: For test case population, test data is used. Also, in order not to break the layout of the test case list table, I had to put the mocked population outside of the table for now.

File: src/components/testCaseList/TestCase.test.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { render, screen } from "@testing-library/react";
+import { MemoryRouter } from "react-router-dom";
+import TestCaseComponent from "./TestCase";
+
+describe("TestCase component", () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it("should render test case status as executed", async () => {
```
[the remark this answers is not in the document]
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.test.tsx]: @jkotanchik-SB Thanks a lot, that makes sense. I just pushed in the change.

File: src/components/testCaseList/TestCaseList.tsx [hunk 2]
```diff
@@ -3,24 +3,35 @@ import tw from "twin.macro";
...
+
   useEffect(() => {
     if (!testCases) {
       testCaseService
         .getTestCasesByMeasureId(measureId)
         .then((testCaseList: TestCase[]) => {
+          //MAT-3911: only for mock up purpose
+          const listItems = testCaseList.map((testCase) => {
+            testCase.executionStatus = "NA";
+            lastTestCase = testCase;
```
Comment (jkotanchik-SB) [inline src/components/testCaseList/TestCaseList.tsx:33]: Is `lastTestCase` used anywhere else? If not, can it be removed?

File: src/components/testCaseList/TestCase.tsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import tw from "twin.macro";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faArrowRight, faArrowUp } from "@fortawesome/free-solid-svg-icons";
+import { useNavigate } from "react-router-dom";
+import { Button } from "@madie/madie-components";
+
+const TH = tw.th`p-3 border-b text-left text-sm font-bold uppercase`;
```
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:8]: I think this is unused constant

File: src/components/testCaseList/TestCase.tsx [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+const TD = tw.td`p-3 whitespace-nowrap text-sm font-medium text-gray-900`;
+const EditButton = tw.button`text-blue-600 hover:text-blue-900`;
+
+const TestCase = (props) => {
+  const navigate = useNavigate();
+  const expanded = props.activeItem === props.index;
+  const cls = expanded ? "item-active" : "";
```
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:15]: same here

File: src/components/testCaseList/TestCase.tsx [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  const isExecuteButtonClicked = props.isExecuteButtonClicked;
+
+  const toggleExpandedHandler = (index) => {
+    props.setActiveItem(index);
+  };
+
+  return (
+    <tr
+      tw="border-b"
+      key={props.testCase.id}
```
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:26]: `testCase` is declared above so can be used directly `testCase.id`

File: src/components/testCaseList/TestCase.tsx [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      data-testid={`test-case-row-${props.testCase.id}`}
```
Comment (adongare) [inline src/components/testCaseList/TestCase.tsx:27]: same here `testCase.id`

File: src/components/testCaseList/TestCaseList.tsx (hunk 2 above)
Comment (adongare) [reply src/components/testCaseList/TestCaseList.tsx:33]: Can you pls try removing it and see if it breaks(also linter is suggesting to not update variable directly in useEffect).

File: src/components/testCaseList/TestCaseList.tsx [hunk 7]
```diff
@@ -50,24 +86,19 @@ const TestCaseList = () => {
...
-                        Edit
-                      </Button>
-                    </TD>
-                  </tr>
+                  <TestCaseComponent
+                    testCase={testCase}
+                    children={null}
+                    setActiveItem={setActiveItem}
+                    activeItem={activeItem}
+                    index={testCase.id}
```
Comment (adongare) [inline src/components/testCaseList/TestCaseList.tsx:94]: It is good to have a unique `key` for each item while iterating. Maybe renaming the `index` to `key` would work.

File: src/components/testCaseList/TestCase.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx:8]: @adongare This is removed.

File: src/components/testCaseList/TestCase.tsx (hunk 4 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx:15]: @adongare Removed.

File: src/components/testCaseList/TestCase.tsx (hunk 5 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx:26]: @adongare Done.

File: src/components/testCaseList/TestCaseList.tsx (hunk 7 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCaseList.tsx]: @adongare Done.

File: src/components/testCaseList/TestCaseList.tsx (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCaseList.tsx]: @adongare It's removed.

**Review by adongare — APPROVED** [review]

File: src/components/testCaseList/TestCase.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.test.tsx]: @jkotanchik-SB TestCase.tsx had a couple of variables passed in before, one of them was executionStatus, and that was why I had that test case before, but my later version was changed and the executionStatus was not a parameter any more, therefore I removed the test case. Please let me know if I could mark this conversation resolved. Thanks!

File: src/components/testCaseList/TestCase.tsx (hunk 6 above)
Comment (sb-cecilialiu) [reply src/components/testCaseList/TestCase.tsx]: This was resolved so I'll mark it as Resolved.

**Pull request merged**